src/foam/nanos/crunch/ExpireUserCapabilityJunctionsCron.java

```diff
@@ -57,12 +60,21 @@ public void execute(X x) {
       if ( activeJunction.getIsInGracePeriod() || activeJunction.getGracePeriod() <= 0 ) {
         activeJunction.setStatus(CapabilityJunctionStatus.EXPIRED);
         activeJunction.setIsExpired(true);
+        userCapabilityJunctionDAO.put(activeJunction);
       } else {
         activeJunction.setIsInGracePeriod(true);
+
+        // update ucj data as renewable
+        FObject data = (FObject) activeJunction.getData();
+        if ( data != null || data instanceof RenewableData ) {
+          RenewableData renewableData = (RenewableData) data;
+          renewableData.setRenewable(true);
+          activeJunction.setData(renewableData);
```

Comment:
wouldn't we want to expire this activeJunction?
Here is where a ucj is past expire date and not in gracePeriod .. but still granted.

Reply:
No, the expiry logic is done in the if statement. The else statement is for the gracePeriod flow.

Reply:
ok so after another look - so ya your right, we wouldn't expire here...but this is where we should be 
activeJunction.setIsInGracePeriod(true);